workers/unity/Packages/com.improbable.gdk.core/Tests/Editmode/GameObjectRepresentation/SpatialOSBehaviourLibraryTests.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System.CodeDom;
+using System.Linq;
+using System.Text.RegularExpressions;
+using NUnit.Framework;
+using UnityEngine;
+using UnityEngine.TestTools;
+using Object = UnityEngine.Object;
+
+namespace Improbable.Gdk.Core.EditmodeTests
+{
+    [TestFixture]
+    public class SpatialOSBehaviourLibraryTests
+    {
+        private const uint ComponentId1 = 1337;
+        private const uint ComponentId2 = 1338;
+
+        [ReaderInterface]
+        [ComponentId(ComponentId1)]
+        private interface DummyReader
+        {
+        }
+
+        [WriterInterface]
+        [ComponentId(ComponentId1)]
+        private interface DummyWriter1
+        {
+        }
+
+        [WriterInterface]
+        [ComponentId(ComponentId2)]
+        private interface DummyWriter2
+        {
+        }
+
+        private class SingleReaderBehaviour : MonoBehaviour
```

Comment:
How about a test for a Reader + Writer?

Reply:
Do you mean a Reader + Writer for the same component (which should give an error just like two readers of the same type), or two separate components? If it's separate, I thought it was really not that different from two different readers. If it's the same, I could see an argument for that.